src/Mod/Draft/draftobjects/dimension.py

```diff
@@ -0,0 +1,235 @@
+    """makeDimension(p1,p2,[p3]) or makeDimension(object,i1,i2,p3)
+    or makeDimension(objlist,indices,p3): Creates a Dimension object with
+    the dimension line passign through p3.The current line width and color
+    will be used. There are multiple  ways to create a dimension, depending on
+    the arguments you pass to it:
+    - (p1,p2,p3): creates a standard dimension from p1 to p2
+    - (object,i1,i2,p3): creates a linked dimension to the given object,
+    measuring the distance between its vertices indexed i1 and i2
+    - (object,i1,mode,p3): creates a linked dimension
+    to the given object, i1 is the index of the (curved) edge to measure,
+    and mode is either "radius" or "diameter".
+    """
+    if not App.ActiveDocument:
+        App.Console.PrintError("No active document. Aborting\n")
+        return
+    obj = App.ActiveDocument.addObject("App::FeaturePython","Dimension")
+    LinearDimension(obj)
+    if App.GuiUp:
+        ViewProviderLinearDimension(obj.ViewObject)
+    if isinstance(p1,App.Vector) and isinstance(p2,App.Vector):
+        obj.Start = p1
+        obj.End = p2
+        if not p3:
+            p3 = p2.sub(p1)
+            p3.multiply(0.5)
+            p3 = p1.add(p3)
+    elif isinstance(p2,int) and isinstance(p3,int):
+        l = []
+        idx = (p2,p3)
+        l.append((p1,"Vertex"+str(p2+1)))
+        l.append((p1,"Vertex"+str(p3+1)))
+        obj.LinkedGeometry = l
+        obj.Support = p1
+        p3 = p4
+        if not p3:
+            v1 = obj.Base.Shape.Vertexes[idx[0]].Point
+            v2 = obj.Base.Shape.Vertexes[idx[1]].Point
+            p3 = v2.sub(v1)
+            p3.multiply(0.5)
+            p3 = v1.add(p3)
+    elif isinstance(p3,str):
+        l = []
+        l.append((p1,"Edge"+str(p2+1)))
+        if p3 == "radius":
+            #l.append((p1,"Center"))
+            if App.GuiUp:
+                obj.ViewObject.Override = "R $dim"
+            obj.Diameter = False
+        elif p3 == "diameter":
+            #l.append((p1,"Diameter"))
+            if App.GuiUp:
+                obj.ViewObject.Override = "Ø $dim"
+            obj.Diameter = True
+        obj.LinkedGeometry = l
+        obj.Support = p1
+        p3 = p4
+        if not p3:
+            p3 = p1.Shape.Edges[p2].Curve.Center.add(App.Vector(1,0,0))
+    obj.Dimline = p3
+    if hasattr(App,"DraftWorkingPlane"):
+        normal = App.DraftWorkingPlane.axis
+    else:
+        normal = App.Vector(0,0,1)
+    if App.GuiUp:
+        # invert the normal if we are viewing it from the back
+        vnorm = gui_utils.get3DView().getViewDirection()
+        if vnorm.getAngle(normal) < math.pi/2:
+            normal = normal.negative()
+    obj.Normal = normal
+    if App.GuiUp:
+        gui_utils.format_object(obj)
+        gui_utils.select(obj)
+
+    return obj
+
+class LinearDimension(DimensionBase):
+    """The Draft Dimension object"""
+    def __init__(self, obj):
+        DimensionBase.__init__(self,obj,"Dimension")
```

Comment:
The better way is to call `super().__init__(self,obj,"Dimension")`.

We prefer to avoid repeating the parent class in case it changes.